src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -597,6 +597,25 @@ private PackageRestoreInputs DetermineRestoreInputs()
             return packageRestoreInputs;
         }
 
+        private bool IsSolutionOrProjectFile(string fileName)
+        {
+            if (!String.IsNullOrEmpty(fileName) && File.Exists(fileName))
+            {
+                string extension = Path.GetExtension(fileName);
+                string lastFourCharacters = string.Empty;
+
+                if (extension.ToLowerInvariant().Contains("proj"))
```

Comment:
This can be stored as a bool, the actual value here isn't needed from what I see.